model/UserModel.ts

```diff
@@ -1,5 +1,153 @@
+import type { UidIdentifier } from "firebase-admin/auth";
+
 export class UserModel {
-    /**
-     * Model for the user
-     */
-}
\ No newline at end of file
+    //private fields for the user's statistics
+
+    //name of the user
+    private _userName: string = '';
+
+    //current streak of the user
+    private _currentStreak: number = 0;
+    //max streak of the user
+    private _maxStreak: number = 0;
+
+    //average rank of the user
+    private _averageRank: number = 0;
+    //array of ranks of the user
+    private _ranks: number[] = [];
+
+    //average number of guesses of the user
+    private _averageGuesses: number = 0;
+    //array of number of guesses of the user
+    private _guesses: number[] = [];
+
+    //average time taken by the user
+    private _averageTime: number = 0;
+    //array of times taken by the user
+    private _times: number[] = [];
+
+    //number of times the user has played
+    private _timesPlayed: number = 0;
+
+    constructor(userName: string) {
+        this._userName = userName;
+        this._currentStreak = 12;
+        this._maxStreak = 14;
+        this._ranks = [36, 2, 12, 44, 17];
+        this._averageRank = this._ranks.reduce((a, b) => a + b, 0) / this._ranks.length;
+        this._guesses = [7, 1, 22, 9, 3];
+        this._averageGuesses = this._guesses.reduce((a, b) => a + b, 0) / this._guesses.length;
+        this._times = [14, 28, 37, 48, 50];
+        this._averageTime = this._times.reduce((a, b) => a + b, 0) / this._times.length;
+        this._timesPlayed = 5;
+    }
+
+    //getter for userName
+    get userName(): string {
+        return this._userName;
+    }
+
+    //getter and setter for currentStreak
+    get currentStreak(): number {
+        return this._currentStreak;
+    }
+    set currentStreak(value: number) {
+        this._currentStreak = value;
+    }
+
+    //getter and setter for maxStreak
+    get maxStreak(): number {
+        return this._maxStreak;
+    }
+    set maxStreak(value: number) {
+        this._maxStreak = value;
+    }
+
+    //getter and setter for averageRank
+    get averageRank(): number {
+        return this._averageRank;
+    }
+    set averageRank(value: number) {
+        this._averageRank = value;
+    }
+
```

Comment:
I don't think setters are needed, once it is constructed you don't change the value. (comment for all setters even though I selected only two here)

Reply:
Since we plan to store those data in our User DB, we should be able to update them when needed.
It will be clarified with issue #29.

Reply:
Then do a method updateData base, not all these setters

Reply:
Yes good point, I will implement this in issue #29

Reply:
I agree with @roxannecvl, I don't really see the point of having both getters and `updateData`. I would prefer a single method `updateDate` to avoid boilerplate code.